### PR TITLE
fix incorrect beta apiVersion for csidriver used in older kubernetes versions

### PR DESCRIPTION
**Pull request by thojkooi**
# Describe what this PR does #

Ran into this when installing this on a 1.17 cluster.
This MR fixes a typo for installing the CSIDriver resource in Kubernetes versions before 1.18. This should have been `storage.k8s.io/v1beta1`.
See: https://kubernetes-csi.github.io/docs/csi-driver-object.html#changes-from-alpha-to-beta

## Is there anything that requires special attention ##

n/a

## Related issues ##

n/a

## Future concerns ##

n/a

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Comment by Madhu-1** [comment]
> * [new branch]          devel      -> origin/devel
commitlint --verbose --from origin/devel
⧗   input: fix incorrect beta apiVersion for csidriver used in older kubernetes versions
See: https://kubernetes-csi.github.io/docs/csi-driver-object.html#changes-from-alpha-to-beta
Signed-off-by: Thomas Kooi <t.j.kooi@avisi.nl>
✖   header must not be longer than 72 characters, current length is 77 [header-max-length]
✖   subject may not be empty [subject-empty]
⚠   body's lines must not be longer than 80 characters [body-max-line-length]

✖   found 2 problems, 1 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
make: *** [Makefile:151: commitlint] Error 1
make: *** [Makefile:189: containerized-test] Error 2
Error: Process completed with exit code 2.

@thojkooi can you please fix the commitlint

**Comment by Rakshith-R** [comment]
hey @thojkooi , Thanks for the pr ! 

Can you modify ur commit message to something similar to 
`cleanup: fix incorrect beta apiVersion for csidriver`
Refer to https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages for more details . 

Otherwise, this pr look good to me !

**Comment by Rakshith-R** [comment]
@thojkooi please fix the following too 
```
⧗   input: fix incorrect beta apiVersion for csidriver used in older kubernetes versions

See: https://kubernetes-csi.github.io/docs/csi-driver-object.html#changes-from-alpha-to-beta

Signed-off-by: Thomas Kooi <t.j.kooi@avisi.nl>
✖   header must not be longer than 72 characters, current length is 77 [header-max-length]
✖   subject may not be empty [subject-empty]
⚠   body's lines must not be longer than 80 characters [body-max-line-length]
```
ideally, shorter header message and truncated body lines like
 ```
 cleanup: fix incorrect beta apiVersion for csidrive
 
 This MR fixes a typo for installing the CSIDriver resource
 in Kubernetes versions before 1.18. 
This should have been storage.k8s.io/v1beta1.

Signed-off-by: Thomas Kooi <t.j.kooi@avisi.nl> 
 ```
Thanks

**Comment by thojkooi** [comment]
Sorry, should be fixed now! Thanks for the help.

**Review by yati1998 — APPROVED** [review]
LGTM

**Review by Madhu-1 — APPROVED** [review]

**Review by Yuggupta27 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

More changes were merged recently, including better failure handling for some CI issues.

**Comment by Madhu-1** [comment]
@thojkooi can you please provide access to allow changes to this pull request https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork. so Mergify bot can rebase and merge this PR

**Review by pkalever — CHANGES_REQUESTED** [review]
Could you also fix the matching comments?
```
examples/cephfs/snapshot.yaml:# v1betav1:
examples/cephfs/snapshotclass.yaml:# v1betav1:
examples/rbd/snapshot.yaml:# v1betav1:
examples/rbd/snapshotclass.yaml:# v1betav1:
```

**Comment by nixpanic** [comment]
@thojkooi this project uses mergify.io for rebasing all PRs on top of the branch before they get merged. That means, PRs get only merged when there have been no recent changes to the base branch (`devel`), and all required tests have passed.

For the Mergify-bot, it is required that you allow the maintainers of this project to update the branch that was used to create this PR. You can enable that on the right side of the PR, there is a checkbox like this:

image

If for whatever reason you can not accept/enable that, let us know and we can create a new PR with your change for you.

Thanks!

**Comment by nixpanic** [comment]
@mergifyio rebase

Checking if the permissions of the PR have been corrected. When mergify can rebase the PR, it will merge it once tests have passed.

**Comment by nixpanic** [comment]
#2310 is a replacement for this PR. Once it is merged, this one will automatically be closed.

**Pull request closed without merging**
